apps/studio/src/features/dashboard/components/FolderSettingsModal/FolderSettingsModal.tsx

```diff
@@ -158,13 +158,9 @@ const SuspendableModalContent = ({
                   <Input
                     placeholder="This is a url for your folder"
                     {...field}
+                    maxLength={MAX_FOLDER_PERMALINK_LENGTH}
                     onChange={(e) => {
-                      onChange(
-                        generateResourceUrl(e.target.value).slice(
-                          0,
-                          MAX_FOLDER_PERMALINK_LENGTH,
-                        ),
-                      )
+                      onChange(generateResourceUrl(e.target.value))
```

Comment:
issue: wait no, we still need the slice (either here or in generateResourceUrl) cos the changes can be coming from outside of the user input in this field.

Reply:
how can the change come from uotside user input ah?

Reply:
My bad, got confused, so this is fine.